rts/motoko-rts-tests/src/gc/heap.rs

```diff
@@ -217,15 +227,15 @@ impl MotokoHeapInner {
             &object_addrs,
             continuation_table_ptr_offset,
             static_heap_size_bytes + dynamic_heap_size_without_continuation_table_bytes,
-            &mut heap[..static_heap_size_bytes],
+            &mut heap[realign..static_heap_size_bytes + realign],
         );
 
         MotokoHeapInner {
             heap: heap.into_boxed_slice(),
-            heap_base_offset: static_heap_size_bytes,
-            heap_ptr_offset: total_heap_size_bytes,
-            static_root_array_offset: 0,
-            continuation_table_ptr_offset: continuation_table_ptr_offset,
+            heap_base_offset: static_heap_size_bytes + realign,
+            heap_ptr_offset: total_heap_size_bytes + realign,
+            static_root_array_offset: realign,
+            continuation_table_ptr_offset: continuation_table_ptr_offset + realign,
```

Comment:
This shifts the entire static heap, but that's not what we do in the compiler. In the compiler the padding is added at the end of the static heap. Should they be in sync?

Reply:
Good observation, but not really relevant for the inner workings, I think. I'll try to only realign the dynamic portion. Hopefully that won't break relative references from the dynamic heap into the static one (they are never followed by GC, but they might get accessed otherwise?).

Reply:
Hmm, I tried to do this, but the necessary gap needs to go into the `heap_size` computation and which is needed for the `vec!` allocation, which then determines the necessary gap size. The circularity introduced would make this very fragile and much harder to understand.

So I won't do this now.

That said, I could map
``` rust
        |gap| heap_size_for_gc(
            gc,
            static_heap_size_bytes + gap,
            dynamic_heap_size_bytes,
            map.len(),
        )
```
over `[0, 4, 8, .. 28]` and get the heap size from there. But this speculative approach is also not worth it, IMO.